integration/src/test/java/com/fitgoal/NotificationServiceIntegrationTest.java

```diff
@@ -0,0 +1,93 @@
+package com.fitgoal;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Response;
+
+import org.junit.Test;
+
+import com.fitgoal.api.domain.Recipient;
+import com.fitgoal.api.domain.UserVerification;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class NotificationServiceIntegrationTest extends AbstractIntegrationTest {
+
+    @Test
+    public void registerTest() {
+        UserVerification userVerification = new UserVerification("test@test.com", "verification_link");
+        Response response = APP_EXTENSION.client()
+                .target(resourcePath)
+                .path("/register")
+                .request()
+                .post(Entity.json(userVerification));
+
+        assertThat(response).isNotNull();
+        assertThat(response.getStatus()).isEqualTo(204);
+    }
+
+    @Test
+    public void registerIllegalArgumentsTest() {
```

Comment:
Proposition, you can create a list of all the possible cases when registration will fail and loop through them in one test named something like `register_whenUserDetailsAreIncorrect_expect422`. Because from this and the one below tests we can't verify the exact problem that occurred under the hood.
So it's just a thing to keep in mind, there is no need to refactor the tests.